### PR TITLE
Rename, extend, document and test `all_monomials`

### PR DESCRIPTION
**Pull request by joschmitt**
* I renamed `all_monomials`, which gives an iterator over all monomials of a polynomial ring of a given degree, to the more appropriate `monomials_of_degree` and exported (+test, document) this function.
* I added the variant `monomials_of_degree(::MPolyRing, ::Int, ::UnitRange)` which only works on a subset of the variables specified by the range and which should cover the use case in #2823 .
* I decided to make it throw an error for non-standard graded rings. So far it works for those rings by ignoring the grading. @HechtiDerLachs is this fine for you? I don't want to break your code or use case.

### CODE REVIEW

File: src/InvariantTheory/iterators.jl [hunk 1]
```diff
@@ -1,23 +1,85 @@
...
+ y*z^2
+ z^3
+
+julia> sum(monomials_of_degree(R, 4, 1:2))
+x^4 + x^3*y + x^2*y^2 + x*y^3 + y^4
+
+julia> sum(monomials_of_degree(R, 4, [1, 3]))
+x^4 + x^3*z + x^2*z^2 + x*z^3 + z^4
+```
+"""
```
Comment (thofma) [inline src/InvariantTheory/iterators.jl:39]:
```suggestion
"""
monomials_of_degree

```

Comment (thofma) [reply src/InvariantTheory/iterators.jl:39]: Then the docstring is attached to the function and not one particular method.

**Comment by thofma** [comment]
Looking at the coverage, there seem to be a few (changed) lines missing. Did you test this locally or do we hope that nothing breaks?

**Comment by HechtiDerLachs** [comment]
Fine with me. Thanks for the notification!

File: src/InvariantTheory/iterators.jl (hunk 1 above)
Comment (joschmitt) [reply src/InvariantTheory/iterators.jl:39]: Thanks, I did not know about this.

**Comment by joschmitt** [comment]
> Looking at the coverage, there seem to be a few (changed) lines missing. Did you test this locally or do we hope that nothing breaks?

I added more tests now so that the actual iterator should be covered completely. Otherwise it is the renaming of a so far not exported function, so I hope it is fine?

**Comment by thofma** [comment]
> I added more tests now so that the actual iterator should be covered completely. Otherwise it is the renaming of a so far not exported function, so I hope it is fine?

Sure. I will merge once CI shows green.

**Review by thofma — APPROVED** [review]

**Pull request merged**